tower-balance/examples/demo.rs

```diff
@@ -89,46 +88,64 @@ fn main() {
     rt.shutdown_on_idle().wait().unwrap();
 }
 
-fn gen_disco() -> Disco {
-    use self::Change::Insert;
+type Error = Box<::std::error::Error + Send + Sync>;
 
-    let mut changes = VecDeque::new();
-    for (i, latency) in MAX_ENDPOINT_LATENCIES.iter().enumerate() {
-        changes.push_back(Insert(i, DelayService(*latency)));
-    }
+fn gen_disco() -> impl Discover<
+    Key = usize,
+    Error = impl Into<Error>,
+    Service = impl Service<Req, Response = Rsp, Error = Error, Future = impl Send> + Send,
+> + Send {
+    tower_discover::ServiceList::new(MAX_ENDPOINT_LATENCIES.iter().map(|latency| {
+        let svc = ServiceFn::new(move |_| {
+            let start = Instant::now();
+            let maxms = u64::from(latency.subsec_nanos() / 1_000 / 1_000)
+                .saturating_add(latency.as_secs().saturating_mul(1_000));
+            let latency = Duration::from_millis(rand::thread_rng().gen_range(0, maxms));
+            let delay = timer::Delay::new(start + latency);
+
+            delay.map(move |_| {
+                let latency = Instant::now() - start;
+                Rsp { latency }
+            })
+        });
 
-    Disco { changes }
+        InFlightLimit::new(svc, ENDPOINT_CAPACITY)
```

Comment:
layer? 😄 or servicebuilder?

Reply:
I will probably update this to `ServiceBuilder` once the misc re-exports / tweaks are applied.